vignettes/value-labels.Rmd

```diff
@@ -16,99 +15,139 @@ knitr::opts_chunk$set(
-it was designed for efficient calculations in linear models, not as a general 
-purpose value labeling system and so is missing important features that the 
-value labels provided by IPUMS require.
+## `labelled` vs. `factor`
+
+Base R already supports the linking of numeric data to categories using
+its `factor` data type. While factors may be more familiar, they were
+designed to support efficient calculations in linear models, not as a
+human-readable labeling system for interpreting and processing data.
+
+Compared to factors, `labelled` vectors have a few properties that make
+them more suitable for working with IPUMS data:
+
+-   They provide more complete contextual information about the meaning
+    of certain values
+-   They don't require that all values be labelled
+-   They don't require values to be assigned to increasing integers
+    starting at 1
 
-Factors only allow for integers to be mapped to a text label, and these integers
-have to be a count starting at 1. This doesn't work for IPUMS data because often
-our variables have specific meanings for the codes. For example, the variable
-`AGE` uses the value to mean the actual age, but does have labels
-for age 0 and the top codes.
+Consider the case of the `AGE` variable. For many IPUMS products, `AGE`
+provides a person's age in years, but certain special values have other
+interpretations:
 
 ```{r}
 head(cps$AGE)
+```
+
+As you can see, the 0 value represents all ages less than 1, and the 90
+and 99 values actually represent ranges of ages. Coercing `AGE` to a
+factor would convert all values of 0 to 1, because factors always assign
+values starting at 1:
 
+```{r}
 cps$AGE_FACTOR <- as_factor(cps$AGE)
-head(cps$AGE_FACTOR)
+
+age0_factor <- cps[cps$AGE == 0, ]$AGE_FACTOR
+
+# The levels look the same
+unique(age0_factor)
+
+# But the values have changed
+unique(as.numeric(age0_factor))
 ```
 
-It may seem like the new `AGE_FACTOR` variable is okay, but it can be very 
-confusing!
+Additionally, because not all values exist in the data, high values,
+like 85, 90, and 99 have been mapped to lower values:
+
 ```{r}
-mean(cps$AGE)
+age85_factor <- cps[cps$AGE == 85, ]$AGE_FACTOR
+  
+unique(as.numeric(age85_factor))
+```
 
-# mean(cps$AGE_FACTOR) # error because data is a factor not numeric
+These different representations lead to inconsistencies in calculated
+values:
 
-mean(as.numeric(cps$AGE_FACTOR)) # A common mistake
+```{r}
+mean(cps$AGE)
 
-# The "more" correct way, but NA because of the text labels
-mean(as.numeric(as.character(cps$AGE_FACTOR)))
+mean(as.numeric(cps$AGE_FACTOR))
 ```
 
-Because the factor variable has to assign values starting at 1, but the
-AGE variable started at 0, most values were 1 higher than they should
-have been. Not all values are 1 higher though, because not all values
-exist in the data, so 85, 90, and 99 are 82, 83 and 84 respectively.
+### Drawbacks of `labelled` variables
```

Comment:
I changed this to "Cautions regarding `labelled` variables", but I'm open to feedback on that.